### PR TITLE
Update RMC NMEA 4.1

### PR DESCRIPTION
**Pull request by bettellam**
Add new item for RMC sentence  NMEA v4.1

### CODE REVIEW

**Comment by nirvn** [comment]
@bettellam , is there something missing here? When we discussed this, you mentioned the parser code needed tweaking.

**Comment by bettellam** [comment]
@nirvn Yes you're right, I made a mistake, the other piece of code ended up in a second patch!!
I'll correct everything!

File: external/nmea/parse.c [hunk 1]
```diff
@@ -415,14 +415,14 @@ int nmea_parse_GPRMC( const char *buff, int buff_sz, nmeaGPRMC *pack )
...
+                     "$G%CRMC,%s,%C,%f,%C,%f,%C,%f,%f,%2d%2d%2d,%f,%C,%C,%C*",
                      &( type ), &( time_buff[0] ),
                      &( pack->status ), &( pack->lat ), &( pack->ns ), &( pack->lon ), &( pack->ew ),
                      &( pack->speed ), &( pack->direction ),
                      &( pack->utc.day ), &( pack->utc.mon ), &( pack->utc.year ),
-                     &( pack->declination ), &( pack->declin_ew ), &( pack->mode ) );
+                     &( pack->declination ), &( pack->declin_ew ), &( pack->mode ), &( pack->navstatus ) );
 
-  if ( nsen != 14 && nsen != 15 )
+  if ( nsen != 14 && nsen != 15 && nsen != 16 )
```
Comment (nyalldawson) [inline external/nmea/parse.c:425]:
```suggestion
  if ( nsen < 14 || nsen > 16 )
```

**Review by nyalldawson — APPROVED** [review]

**Comment by nirvn** [comment]
I vote for backporting to 3.36 and queue for 3.34 LTR. Without this fix, GNSS devices using the NMEA 4.1 spec are not working when switching to high accuracy.

**Comment by nyalldawson** [comment]
@nirvn 

Backporting sounds good

**Comment by bettellam** [comment]
@nyalldawson  @nirvn - Test with receiver using NMEA v4.1 (UNICORECOMM UM982) all OK!

**Pull request merged**